### DISCUSSION TITLE
Support with pnpm install --frozen-lockfile

### DISCUSSION THREAD

**Discussion by grug** [renovatebot/renovate#17550] [Request Help]
Hi there,

In my CI pipelines, I install dependencies with `pnpm install --frozen-lockfile` but that leads to the following error on PRs that are opened from Renovate:

> ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up-to-date with package.json

I've checked and the lockfile is definitely up to date on `main`. I imagine Renovate needs to also update the lockfile along with the version bumps.

Is this currently supported? I've seen a similar issue here: https://github.com/renovatebot/renovate/issues/8471 but unsure if this really reached a resolution.

Do you need a repo that reproduces this? I'm happy to create one for you.

**Comment by rarkins** [comment]
Yes, please create a reproduction. Renovate should be updating the lock file to match any changes made to `package.json`

**Reply by grug** [reply]
Upon further research, I suspect this is a `pnpm` issue and not a Renovate issue: https://github.com/pnpm/pnpm/issues/5144

**Reply by grug** [reply]
I'm just doing a few tests to narrow it down. If I still believe it's a Renovate issue, I'll throw up a reproduction repository for you :)

**Comment by alper** [comment]
Anybody find a solution here?

The settings are exactly the same for us. Renovate manages to update the pnpm-lock.yaml but then when the generated PR hits CI our `pnpm install --frozen-lockfile` rejects it immediately.

I've looked through the code and it looks like it should be installing the dependencies correctly.

**Reply by alper** [reply]
I ran renovate with trace logs, but that does not work because it made node run out of memory.

**Comment by alper** [comment]
You may be running into this issue?

https://github.com/renovatebot/renovate/discussions/38610#discussioncomment-14739419
